htdocs/compta/facture/class/api_invoices.class.php

```diff
@@ -920,6 +920,118 @@ function settounpaid($id)
         return $this->_cleanObjectDatas($this->invoice);
     }
 
+   /**
+     * Create credit note from invoice
+     *
+     * @param   int 	$id            Invoice ID
+     * @url POST    {id}/createCreditNote
+     *
+     * @return  array 	An invoice object
+     *
+     * @throws 200
+     * @throws 304
+     * @throws 401
+     * @throws 404
+     * @throws 500
+     */
+    function createCreditNote($id)
+    {
+        if(! DolibarrApiAccess::$user->rights->facture->creer) {
+                throw new RestException(401);
+        }
+		
+	    $this->db->begin();
+		
+        $result = $this->invoice->fetch($id);
+        if( ! $result ) {
+                throw new RestException(404, 'Invoice not found');
+        }
+
+        if( ! DolibarrApi::_checkAccessToResource('facture', $this->invoice->id)) {
+                throw new RestException(401, 'Access not allowed for login '.DolibarrApiAccess::$user->login);
+        }
+
+		$result = $this->invoice->fetch_thirdparty();
+        if( ! $result ) {
+                throw new RestException(404, 'Thirdparty not found');
+        }
+
+		if (! $object->paye)	// protection against multiple submit
```

Comment:
Var $object seems not defined here.

Reply:
Yes, sorry about that.
Seen that in latest develop this is resolved,
Thanks
Dan